### DISCUSSION TITLE
Triage and Maintain roles in the Collaborators API

### DISCUSSION THREAD

**Discussion by yns88** [community/community#24798] [Apps, API and Webhooks]
<p>GitHub has now launched Triage and Maintain roles out of beta: <a href="https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization" rel="nofollow noopener">https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization</a></p>
<p>However, the collaborator APIs do not support these new roles.</p>
<ul>
<li>
<p>“List Collaborators” only returns “push”, “pull”, and “admin”: <a href="https://developer.github.com/v3/repos/collaborators/#list-collaborators" rel="nofollow noopener">https://developer.github.com/v3/repos/collaborators/#list-collaborators</a></p>
</li>
<li>
<p>“Permission Level” returns “read” for both Read and Triage collaborators, and “write” for both Write and Maintain collaborators: <a href="https://developer.github.com/v3/repos/collaborators/#review-a-users-permission-level" rel="nofollow noopener">https://developer.github.com/v3/repos/collaborators/#review-a-users-permission-level</a></p>
</li>
<li>
<p>“Add Collaborator” has no documentation on the new roles, but it actually does support adding a collaborator with “triage” from my testing: <a href="https://developer.github.com/v3/repos/collaborators/#add-user-as-a-collaborator" rel="nofollow noopener">https://developer.github.com/v3/repos/collaborators/#add-user-as-a-collaborator</a></p>
</li>
</ul>
<p>With the APIs as they are, it is not currently possible to detect the existence of these new roles for collaborators.</p>

**Comment by phtcon** [comment]
<p>It would be mighty handy to be able to see those levels in the GitHub API.</p>
<p>I literally landed on this page because I have a use case for that, and was trying to figure out how to make it work.  I’m working on a tool for auditing the permission settings of repos, and there are cases where developers should have maintain, but not admin.    I’d like to be able to programmatically tell, but currently I can’t.</p>

**Comment by phtcon** [comment]
<p>Note: My colleague just pointed out that we can distinguish between {Write, Maintain} and Admin, which is helpful, but we can’t distinguish between Write and Maintain.   </p>
<p>So, I was imprecise in my previous post, but there is still a gap between what we need and what we have.</p>

**Comment by ac21** [comment, marked as answer]
<p>Not sure if this is helpful for you, but you can determine the Permission level explicitly using the v4 API (GraphQL).  the request would be:</p>
<pre><code>query {
  repository(owner:"yns88", name:"Rogue-Operative") {
    viewerPermission
  }
}
</code></pre>
<p>which would return:</p>
<pre><code>{
  "data": {
    "repository": {
      "viewerPermission": "ADMIN"
    }
  }
}
</code></pre>
<p>additional details are here:<br>
<a href="https://developer.github.com/v4/object/repository/" rel="nofollow noopener">https://developer.github.com/v4/object/repository/</a></p>

**Comment by gabriel-kohen-by** [comment]
<p>Unfortunatly I could not find them in the REST v3 API (Or whatever is the latest).</p>
<p>Any idea when these will turn up as we’re trying to automate the process of creating repos?</p>
